### PR TITLE
Add FightersDestroyedEvent to combat log.

### PR DESCRIPTION
**Pull request by LGM-Doyle**
FightersDestroyedEvent aggregates all fighter destruction by empire.

This addresses the comment in issue #1334, to make this change to the combat log.

### CODE REVIEW

File: combat/CombatEvents.cpp [hunk 1]
```diff
@@ -935,6 +935,110 @@ void FighterLaunchEvent::serialize<freeorion_xml_iarchive>(freeorion_xml_iarchiv
...
+    events()
+{}
+
+void FightersDestroyedEvent::AddEvent(int target_empire_)
+{ events[target_empire_] += 1; }
+
+std::string FightersDestroyedEvent::DebugString() const {
+    std::stringstream ss;
+    ss << "FightersDestroyedEvent: ";
+    for (const auto& index_and_event: events) {
```
Comment (geoffthemedio) [inline combat/CombatEvents.cpp]: space b4 :

File: default/stringtables/en.txt [hunk 2]
```diff
@@ -4108,7 +4108,7 @@ Combat at %1% on turn %2%:
 #     acted).
 # %5% the draw number (a draw is the acting of a single combat unit).
 ENC_COMBAT_ATTACK_STR
-%1% attacks %2% and does %3% damage
+%1% attacks %2% and does %3% damage.
```
Comment (geoffthemedio) [inline default/stringtables/en.txt]: unrelated / independent formatting / style change?

**Comment by MatGB** [comment]
`
/home/matgb/Freeorion/freeorion/UI/FleetWnd.cpp: At global scope:
/home/matgb/Freeorion/freeorion/UI/FleetWnd.cpp:425:10: warning: ‘{anonymous}::temp_bool’ defined but not used [-Wunused-variable]
     bool temp_bool = RegisterOptions(&AddOptions);
          ^
client/human/CMakeFiles/freeorion.dir/build.make:710: recipe for target 'client/human/CMakeFiles/freeorion.dir/__/__/UI/FleetWnd.cpp.o' failed
make[2]: *** [client/human/CMakeFiles/freeorion.dir/__/__/UI/FleetWnd.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
/home/matgb/Freeorion/freeorion/UI/EncyclopediaDetailPanel.cpp:63:10: warning: ‘{anonymous}::temp_bool’ defined but not used [-Wunused-variable]
     bool temp_bool = RegisterOptions(&AddOptions);
          ^
/home/matgb/Freeorion/freeorion/UI/GalaxySetupWnd.cpp:55:10: warning: ‘{anonymous}::temp_bool’ defined but not used [-Wunused-variable]
     bool temp_bool = RegisterOptions(&AddOptions);
          ^
CMakeFiles/Makefile2:493: recipe for target 'client/human/CMakeFiles/freeorion.dir/all' failed
make[1]: *** [client/human/CMakeFiles/freeorion.dir/all] Error 2
Makefile:149: recipe for target 'all' failed
make: *** [all] Error 2
`
This branch no changes.

**Comment by geoffthemedio** [comment]
> This branch no changes.

Look all irrelevant to these changes...

**Comment by MatGB** [comment]
True, just had the same thing in Master, will post an Issue with more of the log (I hadn't scrolled back there are several pages of errors).

File: combat/CombatEvents.cpp [hunk 3]
```diff
@@ -935,6 +935,110 @@ void FighterLaunchEvent::serialize<freeorion_xml_iarchive>(freeorion_xml_iarchiv
...
+    auto num_events_remaining = events.size();
+    std::stringstream ss;
+
+    // Use show_events_for_empire to show events in this order: viewing empire,
+    // ALL_EMPIRES and then the remainder.
+    auto show_events_for_empire =
+        [&ss, &num_events_remaining, &events_to_show, &viewing_empire_id]
+        (boost::optional<int> show_empire) {
+            int count;
+            int target_empire;
```
Comment (geoffthemedio) [inline combat/CombatEvents.cpp]: Add _id to names of show empire and target empire

**Pull request merged**